strum_tests/tests/enum_try_as.rs

```diff
@@ -0,0 +1,44 @@
+use strum::EnumTryAs;
+
+#[derive(EnumTryAs)]
+enum Foo {
```

Comment:
Let's add a unit like variant and a c-struct like variant to make sure we cover all the cases.